### ISSUE TITLE
take the routes file as a param instead of hardcoded?

### ISSUE THREAD

**Issue by ianstormtaylor** [jaredpalmer/after.js#24]
Just wondering... it seems like from reading the docs that After always takes a `src/_routes.js` file as input, and doesn't care about the rest of the directory structure other than that? If so, it seems like it would be awesome if it actually took that file's path as an input similar to other bundlers, instead of enforcing any directory structure at all, like:

```
after ./src/_routes.js
```

Or, up to the user at the point...

```
after ./src/routes.js
after ./browser/index.js
after ./routes.js
...
```

I might have missed other reasons that After enforces a specific directory structure though in my quick reading, so feel free to close if so.

**Comment by ianstormtaylor** [comment]
Ah, it looks like it does the same thing for the "document" feature. But that seems like it could be solved the same way...

```
after ./src/index.js --document=./src/document.js
```

That way there's no hard enforcement of directory structures. Just an idea!

**Comment by jaredpalmer** [comment]
v cool. this would make the tool awesome for demos and like single page docs w/o the overhead of source directory. 

@ianstormtaylor There are already flags in `.env`, how would you feel about putting that there? idk. not as clean. However, i don't really want 2 places for config.

**Comment by ianstormtaylor** [comment]
@jaredpalmer ah, very interesting. I haven't come across projects that use `.env` for configuration like After does, and I haven't really thought about the tradeoffs there vs. using command line flags. Lots of projects are already using it for `NODE_ENV` implicitly in a lot of cases, so it is interesting to see how that might be expanded to be used for all variables. 

My worry thought would be that the environment variables have no easy way of being scoped, so things like `HOST`, `PUBLIC_PATH`, `VERBOSE`, etc. are very possible to collide with existing environment variables that folks are using. (Shouldn't the all be prefixed by `AFTER_`?)

I tend to think of environment variables as app-level, and environment level. And for the most part tooling shouldn't tap into them except for doing environment-specific tweaks. I'd be worried about using After's configuration style personally, because I use app-level env vars with Heroku, and I'd be worried about things clashing over time and getting really hard to follow.

But that's all a bit of a tangent. I'd personally always prefer flags as opposed to env for this sort of configuration, but who knows.

**Comment by ianstormtaylor** [comment]
FWIW, the CI tools I've used (Circle/Travis) right now have horrible support for the UX of editing/defining environment variables, so this is going to add friction. Whereas flags are much easier to spot things that are fixed across environments. And if people need something to be not fixed, they can always use `$VAR` passed to a flag.

**Comment by jaredpalmer** [comment]
Background: After works identically to create-react-app with respect to environment variables. It is the exact same code. After-specific variables are unprefixed HOST, PORT, VERBOSE. User-land variables must have AFTER_ or they will be ignored. 

Directory structure: The directory structure comes from the fact that After.js essentially a fork of Razzle. The way that After.js works is to  merge the user's `src` and After.js's `lib` directories and then point a modified version of Razzle's config at that. This was just to keep things easy. However, I think it'd be possible to move things around. Since After.js is already required at runtime, a lot of the path issues could be avoided if we published the rest of `lib`'s files in the root of the directory (like `babel.js`, `asyncComponent.js` etc.). We could then guarantee that these are imported properly. The only tricky one is getting the correct require path to the user's `routes.js`. I think we could do a readFile and string replace. 🤔

**Comment by ianstormtaylor** [comment]
@jaredpalmer ah interesting, I have no experience with `create-react-app`, so that's totally new to me. Good to know though!
